### PR TITLE
Parallel CI

### PR DESCRIPTION
**Pull request by sideeffffect**

### CODE REVIEW

**Comment by sideeffffect** [comment]
@cquiroz :pray:

**Comment by sideeffffect** [comment]
@cquiroz :pray: 
Sorry for being such a bother

**Comment by sideeffffect** [comment]
@cquiroz  :pray: Once more and then we should be good.

File: project/plugins.sbt [hunk 1]
```diff
@@ -20,3 +20,5 @@ addSbtPlugin("io.github.davidgregory084" % "sbt-tpolecat" % "0.1.16")
 addSbtPlugin("com.geirsson" % "sbt-ci-release" % "1.5.7")
 
 addSbtPlugin("ch.epfl.lamp" % "sbt-dotty" % "0.5.5")
+
+libraryDependencies += "org.snakeyaml" % "snakeyaml-engine" % "2.3"
```
Comment (cquiroz) [inline project/plugins.sbt:24]: I guess this won't propagate to the clients, only for the build?

File: tests/shared/src/test/scala/org/threeten/bp/TestClock.scala [hunk 2]
```diff
@@ -45,7 +45,8 @@ object TestClock {
 
     def getZone: ZoneId = zone
 
-    def withZone(timeZone: ZoneId): Clock = new TestClock.MockInstantClock(millis, timeZone)
+    override def withZone(timeZone: ZoneId): Clock =
```
Comment (cquiroz) [inline tests/shared/src/test/scala/org/threeten/bp/TestClock.scala:48]: Is this required for java 17?

File: build.sbt [hunk 3]
```diff
@@ -1,9 +1,22 @@
 import org.scalajs.linker.interface.ModuleSplitStyle
-import sbtcrossproject.CrossPlugin.autoImport.{ CrossType, crossProject }
+import sbtcrossproject.CrossPlugin.autoImport.{CrossType, crossProject}
 import sbt._
 import sbt.io.Using
 
-val scalaVer                = "3.1.0"
+val versions: Map[String, String] = {
```
Comment (cquiroz) [inline build.sbt:6]: I wonder if it is simpler to use `sbt-github-actions` doing it the other way around, generate the yml from the sbt

File: project/plugins.sbt (hunk 1 above)
Comment (sideeffffect) [reply project/plugins.sbt:24]: Yes, exactly. It's out private thing here :wink:

File: tests/shared/src/test/scala/org/threeten/bp/TestClock.scala (hunk 2 above)
Comment (sideeffffect) [reply tests/shared/src/test/scala/org/threeten/bp/TestClock.scala:48]: I don't know how the build passed without this. Not related to Java 17 at all. I had to do this also for Scala Native PR.

File: build.sbt (hunk 3 above)
Comment (sideeffffect) [reply build.sbt:6]: Personally I find this _way_ simpler. There's just one source of truth: the GitHub Action config. No need for any manual code generation at all.

**Review by cquiroz — APPROVED** [review]
Thanks a lot, this seems very good

**Pull request merged**